mlflow/pipelines/steps/predict.py

```diff
@@ -122,18 +132,27 @@ def _run(self, output_directory):
             scored_sdf.write.format("delta").saveAsTable(self.step_config["output_location"])
 
         # predict step artifacts
-        scored_sdf.coalesce(1).write.format("parquet").save(
-            os.path.join(output_directory, _SCORED_OUTPUT_FILE_NAME)
-        )
-
-        scored_size = scored_sdf.count()
-        if scored_size > _MAX_PROFILE_SIZE:
-            sample_percentage = _MAX_PROFILE_SIZE / scored_size
-            scored_sdf = scored_sdf.sample(sample_percentage)
+        if databricks_utils.is_in_databricks_runtime():
+            dbfs_path = os.path.join(
+                ".mlflow",
+                _get_execution_directory_basename(self.pipeline_root),
+                _SCORED_OUTPUT_FOLDER_NAME,
+            )
+            shutil.rmtree("/dbfs/" + dbfs_path, ignore_errors=True)
+            scored_sdf.coalesce(1).write.format("parquet").save(dbfs_path)
+            _logger.info("Moving artifact from DBFS to driver disk")
+            shutil.copytree(
+                "/dbfs/" + dbfs_path, os.path.join(output_directory, _SCORED_OUTPUT_FOLDER_NAME)
+            )
+            shutil.rmtree("/dbfs/" + dbfs_path)
+        else:
+            scored_sdf.coalesce(1).write.format("parquet").save(
+                os.path.join(output_directory, _SCORED_OUTPUT_FILE_NAME)
+            )
```

Comment:
Can we abstract this logic into a util function e.g. in https://github.com/mlflow/mlflow/blob/master/mlflow/utils/file_utils.py? I can easily see it's used in other places in the future.